lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,188 @@
+                  if (candidateData.isNotEmpty)
+                    Transform.scale(
+                      scale: 2,
+                      child: Container(
+                        decoration: const BoxDecoration(
+                          color: Color(0x33000000),
+                          shape: BoxShape.circle,
+                        ),
+                      ),
+                    ),
+                  if (piece != null)
+                    Draggable(
+                      hitTestBehavior: HitTestBehavior.translucent,
+                      data: piece,
+                      feedback: PieceDragFeedback(
+                        piece: piece,
+                        squareSize: widget.squareSize,
+                        settings: widget.settings,
+                      ),
+                      childWhenDragging: const SizedBox.shrink(),
+                      onDragStarted: () => draggedPieceOrigin = squareId,
+                      onDraggableCanceled: (_, __) {
+                        widget.onDiscardedPiece?.call(squareId);
+                        draggedPieceOrigin = null;
+                      },
+                      child: PieceWidget(
+                        piece: piece,
+                        size: widget.squareSize,
+                        pieceAssets: widget.settings.pieceAssets,
+                      ),
+                    ),
+                ],
+              );
+            },
+            onAcceptWithDetails: (details) {
+              widget.onDroppedPiece?.call(
+                draggedPieceOrigin,
+                squareId,
+                details.data,
+              );
+              draggedPieceOrigin = null;
+            },
+          ),
+        ),
+      );
+    }).toList();
+
+    final background = widget.settings.enableCoordinates
+        ? widget.orientation == Side.white
+            ? widget.settings.colorScheme.whiteCoordBackground
+            : widget.settings.colorScheme.blackCoordBackground
+        : widget.settings.colorScheme.background;
+
+    return SizedBox.square(
+      dimension: widget.size,
+      child: Stack(
+        clipBehavior: Clip.none,
+        children: [
+          if (widget.settings.boxShadow.isNotEmpty ||
+              widget.settings.borderRadius != BorderRadius.zero)
+            Container(
+              clipBehavior: Clip.hardEdge,
+              decoration: BoxDecoration(
+                borderRadius: widget.settings.borderRadius,
+                boxShadow: widget.settings.boxShadow,
+              ),
+              child: background,
+            )
+          else
+            background,
+          ...pieceWidgets,
+        ],
+      ),
+    );
+  }
+}
+
+/// Scaled version of a piece that is currently being dragged.
+///
+/// You can use this to drag pieces onto a [BoardEditor] with the same appearance as when the pieces on the board are dragged.
+class PieceDragFeedback extends StatelessWidget {
```

Comment:
This should be put in a `widgets/drag.dart` file and also used by `Board` in place of: https://github.com/lichess-org/flutter-chessground/blob/b789031bf15c20576879ff366c51ca4bb8f5d32b/lib/src/widgets/board.dart#L696

Reply:
Done - Note that for the `Board`, `_DragaAvatar` seems to use slightly different anchoring than `Draggable`, so that's why the `- const Offset(0.5, 0.5)` is there.